### ISSUE TITLE
[Dev] Status Effects with NearbyCharacters

### ISSUE THREAD

**Issue by ChikoEduardo** [Regalis11/Barotrauma#1189]
This new functionality could use a piece of code to allow armor to block afflictions, like it happens with explosions. Also, I'm not sure if there's a way for it to only target a creature type, either.

```
<StatusEffect type="Always" target="NearbyCharacters" range="500">
        <Conditional SpeciesName="Human" />
		<Affliction identifier="cold" strength="0.03" />
      </StatusEffect>
```

This piece for instance. It applies this affliction to all characters in range but only if a human is also in range. This is useful to implement stuff like "airborne" afflictions that will affect all characters when the specified creature is in range but maybe having a way to also specify the affected creatures or a way to enable armor values to absorb or block afflictions would be really handy.

**Comment by rico6822** [comment]
So you want to create an armor that makes everyone more chilly when there is at least 1 human in range of 500 units. Does it affect monsters as well? Is it going to be a bug in it's early phase?

**Comment by Regalis11** [comment]
I think you could block the afflictions by adding DamageModifiers to the armor (the same way ballistic vests provide protection against the bleeding affliction). 

I think the conditional here should prevent the affliction from being applied to characters other than humans, so I'm guessing it's bugged - I'll take a closer look tomorrow.

@rico6822 I would assume Chiko is trying to add a "cold" affliction to nearby humans, and to add gear that keeps you from getting cold.

**Comment by rico6822** [comment]
>I would assume Chiko is trying to add a "cold" affliction to nearby humans, and to add gear that keeps you from getting cold.

So it makes wearer more resistant to get cold but makes everyone around chilly.

**Comment by Regalis11** [comment]
>So it makes wearer more resistant to get cold but makes everyone around chilly.
No, the issue is about two separate things: statuseffects that only target specific kinds of nearby characters, and wearables that protect against afflictions.

**Comment by rico6822** [comment]
> > So it makes wearer more resistant to get cold but makes everyone around chilly.
> 
> No, the issue is about two separate things: statuseffects that only target specific kinds of nearby characters, and wearables that protect against afflictions.

Ok.

**Comment by ChikoEduardo** [comment]
Oh, I've attached DamageModifiers to the diving suit but it seems this applies the afflictions directly into characters, like legacy syringes. I noticed because I made a lot of items in ruins have a "cold AoE" as a way to simulate cold in ruins but it goes thru diving suits even when the armor value is set to 0. The value blocks "cold" explosions completely so I know it works with attacks and explosions.

Also, that was just an example, rico. :p
